target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c

```diff
@@ -941,8 +941,7 @@ static void rtl83xx_get_strings(struct dsa_switch *ds,
 		return;
 
 	for (int i = 0; i < ARRAY_SIZE(rtl83xx_mib); i++)
-		strncpy(data + i * ETH_GSTRING_LEN, rtl83xx_mib[i].name,
-			ETH_GSTRING_LEN);
+		ethtool_sprintf(&data, "%s", rtl83xx_mib[i].name);
```

Comment:
```suggestion
		ethtool_sprintf(&data[i * ETH_GSTRING_LEN], "%s", rtl83xx_mib[i].name);
```

Reply:
Also, there's no `files-6.1` counterpart to this?

Reply:
Realtek has no 6.1 yet. This change seems wrong.

Reply:
Actually it’s completely wrong now that I look at it.

Reply:
Why even use `ethtool_sprintf` if you're not planning to update the source ptr?

Reply:
Because ethtool_sprintf already does so?

Reply:
But are you sure the `strlen` of each name is `ETH_GSTRING_LEN`?

Reply:
I don't think it matters. ethtool_sprintf uses ETH_GSTRING_LEN internally.

Also, see **https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?h=next-20231213id=4ddc1f1f7339e0a03bbf235a6670d497bbb22fb3**

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/net/ethtool/ioctl.c?h=v6.6.7#n1990